package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -818,23 +842,51 @@ export const MessageInputProvider = <
     }
   };
 
+  const regExcondition = /File (extension \.\w{2,4}|type \S+) is not supported/;
+
+  const mapFileUploadState =
+    (id: string, fileState: string, extraData?: Record<string, unknown>) =>
+    <UploadType extends { id: string }>(prevImageUploads: UploadType[]) =>
+      prevImageUploads.map((imageUpload) => {
+        if (imageUpload.id === id) {
+          return {
+            ...imageUpload,
+            ...extraData,
+            state: fileState,
+          };
+        }
+        return imageUpload;
+      });
```

Comment:
```suggestion
  function getUploadStateAction<UploadType extends ImageUpload | FileUpload>(
    id: string,
    fileState: FileStateTypes,
    extraData: Partial<UploadType> = {},
  ): React.SetStateAction<UploadType[]> {
    return (prevUploads: UploadType[]) =>
      prevUploads.map((imageUpload) => {
        if (prevUploads.id === id) {
          return {
            ...prevUploads,
            ...extraData,
            state: fileState,
          };
        }
        return imageUpload;
      });
  }
```

Much more clarity of the function in my view. Now it is "loosely" typed. Maybe prone to errors in the future.

Reply:
This is very helpful and I understand this better. I edited the function a little bit to aid readability on my end

Reply:
✅